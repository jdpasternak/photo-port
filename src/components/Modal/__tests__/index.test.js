import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Modal from "..";

afterEach(cleanup);

const photo = {
  name: "Park bench",
  category: "landscape",
  description: "A park bench outside",
  index: 1,
};

const mockToggleModal = jest.fn();

describe("Modal component", () => {
  it("renders", () => {
    render(<Modal currentPhoto={photo} onClose={mockToggleModal} />);
  });

  it("snapshot", () => {
    const { asFragment } = render(
      <Modal currentPhoto={photo} onClose={mockToggleModal} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});

describe("click event", () => {
  it("calls onClose handler", () => {
    const { getByText } = render(
      <Modal currentPhoto={photo} onClose={mockToggleModal} />
    );

    fireEvent.click(getByText("Close this modal"));

    expect(mockToggleModal).toHaveBeenCalledTimes(1);
  });
});
