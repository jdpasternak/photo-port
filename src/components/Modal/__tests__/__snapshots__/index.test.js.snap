// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component snapshot 1`] = `
<DocumentFragment>
  <div
    class="modalBackdrop"
  >
    <div
      class="modalContainer"
    >
      <h3
        class="modalTitle"
      >
        Park bench
      </h3>
      <img
        alt="current category"
        src="1.jpg"
      />
      <p>
        A park bench outside
      </p>
      <button
        type="submit"
      >
        Close this modal
      </button>
    </div>
  </div>
</DocumentFragment>
`;
